go/filemanager/pfsmodules/localfile.go

```diff
@@ -0,0 +1,121 @@
+package pfsmodules
+
+import (
+	"bytes"
+	"crypto/md5"
+	"encoding/hex"
+	"io"
+	"os"
+
+	log "github.com/golang/glog"
+)
+
+// FileHandle is a local *os.File with offset.
+type FileHandle struct {
+	F      *os.File
+	Offset int64
+}
+
+// Close closes FileHandle.
+func (f *FileHandle) Close() {
+	if f.F != nil {
+		f.F.Close()
+	}
+}
+
+// Open opens a file.
+func (f *FileHandle) Open(path string, flag int, size int64) error {
+	if flag == WriteOnly ||
+		flag == ReadAndWrite {
+
+		cmd := TouchCmd{
+			Method:   TouchCmdName,
+			Path:     path,
+			FileSize: size,
+		}
+		// attempt to create sized file if it does't exist or
+		// file's size != size
+		if err := localTouch(&cmd); err != nil {
+			return err
+		}
+	}
+
+	fd, err := os.OpenFile(path, flag, 0666)
+	if err != nil {
+		return err
+	}
+
+	f.F = fd
+	return nil
+}
+
+func (f *FileHandle) checkOffset(offset int64) error {
+	if offset != f.Offset {
+		if _, err := f.F.Seek(offset, 0); err != nil {
+			return err
+		}
+		f.Offset = offset
+	}
+	return nil
+}
+
+// ReadChunk loads a chunk at offset with len.
+func (f *FileHandle) ReadChunk(offset int64, len int64) (*Chunk, error) {
```

Comment:
`ReadChunk` is a little confusing: it only calculates checksum, but don't actually load the data into Chunk. If all we need is checksum, maybe rename to `Checksum`? Or if we need other information such as real length, we can call it `loadMeta`.